File: src/main/java/com/enderryno/nuclearcraft/interfaces/PluginBlock.java
package com.enderryno.nuclearcraft.interfaces;

import com.enderryno.nuclearcraft.enums.BlockBehaviour;
import com.enderryno.nuclearcraft.enums.ItemBehaviour;

public interface PluginBlock {
    PluginBlock setID(int id);
    int getID();

    PluginBlock setMinecraftId(String minecraftId);
    String getMinecraftId();

    PluginBlock setCustomBlockId(int customBlockId);
    int getCustomBlockId();

    void placeBlock(double x, double y, double z);
    void removeBlock(double x, double y, double z);

    /*Behaviour type*/
    PluginBlock setBehaviour(BlockBehaviour behaviour);
    BlockBehaviour getBehaviour();


}

File: src/main/java/com/enderryno/nuclearcraft/interfaces/PluginBlock.kt
package com.enderryno.nuclearcraft.interfaces

import com.enderryno.nuclearcraft.enums.BlockBehaviour
import org.bukkit.Location

interface PluginBlock {
    val id: String
    val customModelId: Int
    val minecraftId: String
    fun placeBlock(item: PluginItem, location: Location)
    fun removeBlock(location: Location)

    /*Behaviour type*/
    val behaviour: BlockBehaviour
}
